### ISSUE TITLE
bug: it will drop a valid span batch data when the block height of L2 reached 7134334 on the optimism sepolia network

### ISSUE THREAD

**Issue by thinkAfCod** [a16z/magi#207]
Even though every loop has called the update_safe_head method to put block info of 7134334 into the state.l2_refs, it will drop the span batch with the message 'prev L2 block not found'.

If Magi gets the previous L2 block from the op-geth, it will run correctly.

**Comment by GrapeBaBa** [comment]
@ncitron This is tested on OP Seoplia testnet, 7134334 block not executed correctly.

**Comment by thinkAfCod** [comment]
The l2_refs only caches block info in the range of [safe_head - (max_seq_drift / block_time), safe_head].
In the optimism sepolia testnet, the max_seq_drift is 600, and block_time is 2.
When the safe_head is 7134334, the result is [7134034, 7134334]
Unexpectedly, the span batch is valid which timestamp is 1706070328 and refs to 7133894 but the l2 ref info of 7133894 has been removed from l2_refs when the safe head is 7134334.

**Comment by ncitron** [comment]
Oh this is interesting. I was under the impression that after the sequencer drift you are supposed to start injecting empty blocks so this caching length would be acceptable. We'll need to figure out what the correct number to increase it to is.

@GrapeBaBa do you have any idea what it should be?

**Comment by GrapeBaBa** [comment]
> Oh this is interesting. I was under the impression that after the sequencer drift you are supposed to start injecting empty blocks so this caching length would be acceptable. We'll need to figure out what the correct number to increase it to is.
> 
> @GrapeBaBa do you have any idea what it should be?

No, @thinkAfCod current fixed it by qurey op-geth when can't find it in the cache.
